routing/joint_segment.cpp

```diff
@@ -0,0 +1,45 @@
+#include "routing/joint_segment.hpp"
+
+namespace routing
+{
+
+JointSegment::JointSegment(Segment const & from, Segment const & to)
+{
+  CHECK(from.IsRealSegment(), ("Segment of joints can not be fake. Only through ToFake() method."));
```

Comment:
CHECK(from.IsRealSegment() && to.IsRealSegment(), (/**/))?

Reply:
done